eng/pipelines/templates/jobs/live.tests.yml

```diff
@@ -0,0 +1,197 @@
+parameters:
+- name: Location
+  type: string
+  default: ''
+- name: SubscriptionConfiguration
+  type: string
+  default: $(sub-config-azure-cloud-test-resources)
+- name: ServiceDirectory
+  type: string
+  default: not-specified
+- name: CtestRegex
+  type: string
+  default: .*
+- name: Coverage
+  type: string
+  default: 'enabled'
+- name: CoverageReportPath
+  type: string
+  default: sdk/*/*/*cov_xml.xml
+- name: TimeoutInMinutes
+  type: number
+  default: 60
+- name: DependsOn
+  type: string
+  default: ''
+- name: Matrix
+  type: string
+- name: MaxParallel
+  type: number
+  default: 0
+- name: CloudConfig
+  type: object
+  default: {}
+- name: UsePlatformContainer
+  type: boolean
+  default: false
+
+jobs:
+- job: ValidateLive
+  dependsOn: ${{ parameters.DependsOn }}
+  condition: ne(${{ parameters.Matrix }}, '{}')
+
+  pool:
+    name: $(Pool)
+    vmImage: $(OSVmImage)
+
+  ${{ if eq(parameters.UsePlatformContainer, 'true') }}:
+    # Add a default so the job doesn't fail when the matrix is empty
+    container: $[ variables['Container'] ]
+
+  strategy:
+    maxParallel: ${{ parameters.MaxParallel }}
+    matrix: $[ ${{ parameters.Matrix }} ]
+
+  variables:
+    CMOCKA_XML_FILE: "%g-test-results.xml"
+    CMOCKA_MESSAGE_OUTPUT: "xml"
+    AZURE_ENABLE_STATIC_ANALYSIS: 1
+    BuildArgs: ""
+    WindowsCtestConfig: ""
+    CmakeEnvArg: ""
+    CmakeArgs: ""
+    AZURE_TEST_MODE: "LIVE"
+    AZURE_LOG_LEVEL: "verbose"
+    # Surface the ServiceDirectory parameter as an environment variable so tests can take advantage of it.
+    AZURE_SERVICE_DIRECTORY: ${{ parameters.ServiceDirectory }}
+
+  steps:
+  - checkout: self
+    submodules: recursive
+
+  - template: /eng/common/pipelines/templates/steps/verify-agent-os.yml
+    parameters:
+      AgentImage: $(OSVmImage)
+
+  - template: /eng/common/pipelines/templates/steps/bypass-local-dns.yml
+
+  # Add g++5 repo to ubuntu
+  - bash: sudo sh -c "echo 'deb http://ftp.debian.org/debian/ stretch main' >> /etc/apt/sources.list"
+    displayName: Add g++ 5
+    condition: >-
+      and(
+        succeeded(),
+        contains(variables['OSVmImage'], 'Ubuntu'),
+        contains(variables['CmakeEnvArg'], 'CC=/usr/bin/gcc-5 CXX=/usr/bin/g++-5 cmake')
+      )
+
+  # Install g++5
```

Comment:
Why not combine these two steps?

Reply:
https://github.com/Azure/azure-sdk-for-cpp/issues/3565